### PR TITLE
Story9 list orders

### PR DESCRIPTION
**Pull request by Phildallen**

### CODE REVIEW

File: controllers/getOrdersController.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+            const json = {"orders": result}
+            response.send(json)
+        })
+        .catch((error) => {
+            const message = {"message": error.message, "data": []}
+            response.status(500).send(message)
+        })
+}
+
+module.exports.getOrders = getOrders;
```
Comment (JamesMcGraaw) [inline controllers/getOrdersController.js:17]: blank line at end

File: functions/orderDB-to-json.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        return { "SKU": sku, quantity: Number(quantity) };
+    });
+
+    // const sku = idToSKU(idAndQuantity.id)
+    //
+    return { "SKU": "fred", quantity: idAndQuantity.quantity };
+}
+
+
+module.exports.orderDBToJson = orderDBToJson
```
Comment (JamesMcGraaw) [inline functions/orderDB-to-json.js:19]: blank line at end

File: functions/orderDB-to-json.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const {idToSKU} = require("../functions/id-to-sku");
```
Comment (JamesMcGraaw) [inline functions/orderDB-to-json.js:1]: consistency with the rest of the code - have this as const 
idToSKU = require("../functions/id-to-sku");
then in the code
idToSKU.idToSKU();

File: repositories/getOrdersRepository.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const dbService = require('../db/dbService');
+
+const getOrders = async () => {
+    console.log('Repository: getOrders');
```
Comment (JamesMcGraaw) [inline repositories/getOrdersRepository.js:4]: remove console log

File: repositories/getOrdersRepository.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    const connection = await dbService.createConnection()
+    const results = await connection.query('SELECT `id`, `order_number`, `customer_email`, `products` FROM `orders`;');
```
Comment (JamesMcGraaw) [inline repositories/getOrdersRepository.js:6]: can just have return await connection... etc.

File: config/routes.js [hunk 6]
```diff
@@ -2,16 +2,19 @@ const getProductsController = require('../controllers/getProductsController');
...
 const createOrdersController = require('../controllers/createOrdersController');
 const cancelOrdersController = require('../controllers/cancelOrdersController');
 
+
 const routes = (app) => {
     app.get('/products', getProductsController.getProducts); // Get all products
     app.post('/products', addProductsController.addProducts); // Add a products
     app.put('/products/:SKU', updateProductsController.updateProducts); // Update a product
     app.delete('/products/:SKU', deleteProductsController.deleteProducts); // Delete a product
+    app.get('/orders', getOrdersController.getOrders); // Get all orders
```
Comment (JamesMcGraaw) [inline config/routes.js:17]: can we put the orders together with the other orders routes so its all products in one place and all orders in another

Comment (JamesMcGraaw) [reply config/routes.js:17]: could you also make sure that the order of the controllers above is the same order as the routes too :)

File: functions/orderDB-to-json.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    const formattedString = orderSQL.replace(/\(/g, '').replace(/\)/g, ''); // Remove parentheses
+    const pairs = formattedString.split(',');
+
+    const idAndQuantity = pairs.map(pair => {
+        const [id, quantity] = pair.split(':');
+        const sku = idToSKU(id)
+        return { "SKU": sku, quantity: Number(quantity) };
+    });
+
+    // const sku = idToSKU(idAndQuantity.id)
```
Comment (JamesMcGraaw) [inline functions/orderDB-to-json.js:13]: remove comments

File: functions/orderDB-to-json.js (hunk 3 above)
Comment (JamesMcGraaw) [inline functions/orderDB-to-json.js:1]: In fact is this function even used? I can't see it being called anywhere?

File: services/getOrdersService.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+const getOrdersRepository = require('../repositories/getOrdersRepository');
+
+const getOrders = async () => {
+    console.log('Service: getOrders');
```
Comment (JamesMcGraaw) [inline services/getOrdersService.js:4]: remove console log

File: services/getOrdersService.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    try {
+        const results = await getOrdersRepository.getOrders();
```
Comment (JamesMcGraaw) [inline services/getOrdersService.js:6]: refactor this to something more semantic like "allOrdersQueryResponse"

File: services/getOrdersService.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        let resultsArray = [];
```
Comment (JamesMcGraaw) [inline services/getOrdersService.js:7]: refactor this to something more semantic like "allOrdersArray"

File: services/getOrdersService.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+        let eachOrderObject = {};
+
+        results.forEach((result) => {
+            const orderInfo = result.products.split("(").join("").split(")").join("");
+            const orderArrayHalf = orderInfo.split(",");
+            let resultsObject ={};
+            const eachOrder = [];
+            const keys =["name", "quantity"];
```
Comment (JamesMcGraaw) [inline services/getOrdersService.js:15]: rename this to something more semantic like "productsOrderKeys"

File: services/getOrdersService.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+    try {
+        const results = await getOrdersRepository.getOrders();
+        let resultsArray = [];
+        let eachOrderObject = {};
+
+        results.forEach((result) => {
+            const orderInfo = result.products.split("(").join("").split(")").join("");
+            const orderArrayHalf = orderInfo.split(",");
+            let resultsObject ={};
+            const eachOrder = [];
```
Comment (JamesMcGraaw) [inline services/getOrdersService.js:14]: rename this to something more semantic like "currentOrderArray"

File: services/getOrdersService.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+const getOrders = async () => {
+    console.log('Service: getOrders');
+    try {
+        const results = await getOrdersRepository.getOrders();
+        let resultsArray = [];
+        let eachOrderObject = {};
+
+        results.forEach((result) => {
+            const orderInfo = result.products.split("(").join("").split(")").join("");
+            const orderArrayHalf = orderInfo.split(",");
```
Comment (JamesMcGraaw) [inline services/getOrdersService.js:12]: more semantic naming

File: services/getOrdersService.js [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+            let resultsObject ={};
+            const eachOrder = [];
+            const keys =["name", "quantity"];
+            orderArrayHalf.forEach((entry) => {
```
Comment (JamesMcGraaw) [inline services/getOrdersService.js:16]: more semantic naming of entry?

File: services/getOrdersService.js [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+                eachOrder.push(entry.split(":"));
+                eachOrderObject = eachOrder.map((innerArray) => Object.assign({}, ...innerArray.map((value, index) => ({ [keys[index]]: value}))));
```
Comment (JamesMcGraaw) [inline services/getOrdersService.js:18]: what does the ... do in this line?

Comment (JamesMcGraaw) [reply services/getOrdersService.js:18]: also extra space before [keys[index]] at the end of the line

File: services/getOrdersService.js [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+            })
```
Comment (JamesMcGraaw) [inline services/getOrdersService.js:19]: semicolon

File: services/getOrdersService.js [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+
+            resultsObject.order_number = result.order_number;
+            resultsObject.customer_email = result.customer_email;
+            resultsObject.products = eachOrderObject;
+            resultsArray.push(resultsObject);
+        })
+
```
Comment (JamesMcGraaw) [inline services/getOrdersService.js:26]: remove line break

**Review by JamesMcGraaw — CHANGES_REQUESTED** [review]

File: controllers/getOrdersController.js (hunk 1 above)
Comment (Phildallen) [reply controllers/getOrdersController.js:17]: done

File: controllers/getOrdersController.js [hunk 18]
```diff
@@ -0,0 +1,17 @@
+const getOrdersService = require('../services/getOrdersService');
+
+const getOrders = (request, response) => {
+    console.log('Controller: getProducts');
```
[the remark this answers is not in the document]
Comment (Phildallen) [reply controllers/getOrdersController.js:4]: Gone

File: repositories/getOrdersRepository.js (hunk 5 above)
Comment (Phildallen) [reply repositories/getOrdersRepository.js:6]: done

File: config/routes.js (hunk 6 above)
Comment (Phildallen) [reply config/routes.js:17]: done

File: functions/orderDB-to-json.js [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+    //
+    return { "SKU": "fred", quantity: idAndQuantity.quantity };
```
[the remark this answers is not in the document]
Comment (Phildallen) [reply functions/orderDB-to-json.js:15]: File deleted

File: functions/orderDB-to-json.js (hunk 3 above)
Comment (Phildallen) [reply functions/orderDB-to-json.js:1]: file deleted

File: services/getOrdersService.js (hunk 8 above)
Comment (Phildallen) [reply services/getOrdersService.js:4]: done

File: services/getOrdersService.js (hunk 9 above)
Comment (Phildallen) [reply services/getOrdersService.js:6]: done

File: services/getOrdersService.js (hunk 10 above)
Comment (Phildallen) [reply services/getOrdersService.js:7]: done

File: services/getOrdersService.js [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+    console.log('Service: getOrders');
+    try {
+        const results = await getOrdersRepository.getOrders();
+        let resultsArray = [];
+        let eachOrderObject = {};
+
+        results.forEach((result) => {
+            const orderInfo = result.products.split("(").join("").split(")").join("");
+            const orderArrayHalf = orderInfo.split(",");
+            let resultsObject ={};
```
[the remark this answers is not in the document]
Comment (Phildallen) [reply services/getOrdersService.js:13]: done

File: services/getOrdersService.js (hunk 11 above)
Comment (Phildallen) [reply services/getOrdersService.js:15]: done

File: services/getOrdersService.js (hunk 12 above)
Comment (Phildallen) [reply services/getOrdersService.js:14]: done

File: services/getOrdersService.js [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+
+const getOrders = async () => {
+    console.log('Service: getOrders');
+    try {
+        const results = await getOrdersRepository.getOrders();
+        let resultsArray = [];
+        let eachOrderObject = {};
+
+        results.forEach((result) => {
+            const orderInfo = result.products.split("(").join("").split(")").join("");
```
[the remark this answers is not in the document]
Comment (Phildallen) [reply services/getOrdersService.js:11]: done

File: services/getOrdersService.js (hunk 14 above)
Comment (Phildallen) [reply services/getOrdersService.js:16]: done

File: services/getOrdersService.js (hunk 15 above)
Comment (Phildallen) [reply services/getOrdersService.js:18]: ... is the spread operator. space removed

File: services/getOrdersService.js (hunk 16 above)
Comment (Phildallen) [reply services/getOrdersService.js:19]: done

File: services/getOrdersService.js (hunk 17 above)
Comment (Phildallen) [reply services/getOrdersService.js:26]: done

**Review by JamesMcGraaw — APPROVED** [review]

**Pull request merged**